presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -804,9 +814,183 @@ public void testSchemaEvolution()
         assertUpdate("DROP TABLE test_schema_evolution");
     }
 
+    @Test
+    public void testRcAndTextFormats()
+            throws IOException
+    {
+        getQueryRunner().execute("CREATE TABLE lineitem_ex_partitioned WITH (format = 'ORC', partitioned_by = ARRAY['ds']) AS\n" +
+                "SELECT\n" +
+                "    linenumber,\n" +
+                "    orderkey,\n" +
+                "    partkey,\n" +
+                "    suppkey,\n" +
+                "    quantity,\n" +
+                "    extendedprice,\n" +
+                "    tax,\n" +
+                "    shipinstruct,\n" +
+                "    shipmode,\n" +
+                "    ship_by_air,\n" +
+                "    is_returned,\n" +
+                "    ship_day,\n" +
+                "    ship_month,\n" +
+                "    ship_timestamp,\n" +
+                "    commit_timestamp,\n" +
+                "    discount_real,\n" +
+                "    discount,\n" +
+                "    tax_real,\n" +
+                "    ship_day_month,\n" +
+                "    discount_long_decimal,\n" +
+                "    tax_short_decimal,\n" +
+                "    long_decimals,\n" +
+                "    keys,\n" +
+                "    doubles,\n" +
+                "    nested_keys,\n" +
+                "    flags,\n" +
+                "    reals,\n" +
+                "    info,\n" +
+                "    dates,\n" +
+                "    timestamps,\n" +
+                "    comment,\n" +
+                "    uppercase_comment,\n" +
+                "    fixed_comment,\n" +
+                "    char_array,\n" +
+                "    varchar_array,\n" +
+                "    '2019-11-01' AS ds\n" +
+                "FROM lineitem_ex");
+        try {
+            for (HiveStorageFormat format : ImmutableList.of(RCBINARY, RCTEXT, TEXTFILE)) {
+                assertFileFormat(format);
+            }
+        }
+        finally {
+            assertUpdate("DROP TABLE lineitem_ex_partitioned");
+        }
+    }
+
+    private void assertFileFormat(HiveStorageFormat storageFormat)
+            throws IOException
+    {
+        // Make an ORC table backed by file of some other format
+        QueryRunner queryRunner = getQueryRunner();
+        queryRunner.execute("CREATE TABLE test_file_format_orc WITH (format='ORC', partitioned_by=ARRAY['ds']) AS " +
+                "SELECT * FROM lineitem_ex_partitioned LIMIT 1");
+        queryRunner.execute(format("CREATE TABLE test_file_format WITH (format='%s', partitioned_by=ARRAY['ds']) AS " +
```

Comment:
@bhhari this should go inside the try block; otherwise, if it fails, test_file_format_orc table won't get deleted.